src/calliope/preprocess/data_tables.py

```diff
@@ -50,22 +50,20 @@ class DataTable:
 
     def __init__(
         self,
-        model_config: dict,
         table_name: str,
         data_table: DataTableDict,
         data_table_dfs: dict[str, pd.DataFrame] | None = None,
-        model_definition_path: Path | None = None,
+        model_definition_path: str | Path | None = None,
     ):
         """Load and format a data table from file / in-memory object.
 
         Args:
-            model_config (dict): Model initialisation configuration dictionary.
             table_name (str): name of the data table.
             data_table (DataTableDict): Data table definition dictionary.
             data_table_dfs (dict[str, pd.DataFrame] | None, optional):
                 If given, a dictionary mapping table names in `data_table` to in-memory pandas DataFrames.
                 Defaults to None.
-            model_definition_path (Path | None, optional):
+            model_definition_path (Path, optional):
```

Comment:
Should this be `str | Path | None, optional`?

Reply:
I remember removing some of these due to mypy complaining about incorrect typings, but I am unsure about this one. If running `mypy check` does not result in complaints, either is fine.